---
title: AsposeInvalidDataException
second_title: Aspose.Email for Java API Reference
description: The exception that is thrown when one of the arguments provided to a method is not valid.
type: docs
weight: 55
url: /java/com.aspose.email/asposeinvaliddataexception/
---
**Inheritance:**
java.lang.Object, java.lang.Throwable, java.lang.Exception, java.lang.RuntimeException, com.aspose.ms.System.Exception, [com.aspose.email.AsposeException](../../com.aspose.email/asposeexception)
```
public class AsposeInvalidDataException extends AsposeException
```

The exception that is thrown when one of the arguments provided to a method is not valid.

 1 
## Constructors

| Constructor | Description |
| --- | --- |
| [AsposeInvalidDataException()](#AsposeInvalidDataException--) | Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class. |
| [AsposeInvalidDataException(String message)](#AsposeInvalidDataException-java.lang.String-) | Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a specified error message. |
| [AsposeInvalidDataException(String format, Object[] parameters)](#AsposeInvalidDataException-java.lang.String-java.lang.Object...-) | Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a specified error message. |
| [AsposeInvalidDataException(String message, Throwable innerException)](#AsposeInvalidDataException-java.lang.String-java.lang.Throwable-) | Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a specified error message and a reference to the inner exception that is the cause of this exception. |
| [AsposeInvalidDataException(Throwable innerException)](#AsposeInvalidDataException-java.lang.Throwable-) | Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a reference to the inner exception that is the cause of this exception. |
### AsposeInvalidDataException() {#AsposeInvalidDataException--}
```
public AsposeInvalidDataException()
```


Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class.

### AsposeInvalidDataException(String message) {#AsposeInvalidDataException-java.lang.String-}
```
public AsposeInvalidDataException(String message)
```


Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a specified error message.

**Parameters:**
| Parameter | Type | Description |
| --- | --- | --- |
| message | java.lang.String | The message that describes the error. |

### AsposeInvalidDataException(String format, Object[] parameters) {#AsposeInvalidDataException-java.lang.String-java.lang.Object...-}
```
public AsposeInvalidDataException(String format, Object[] parameters)
```


Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a specified error message.

**Parameters:**
| Parameter | Type | Description |
| --- | --- | --- |
| format | java.lang.String | Message format |
| parameters | java.lang.Object[] | Format parameters |

### AsposeInvalidDataException(String message, Throwable innerException) {#AsposeInvalidDataException-java.lang.String-java.lang.Throwable-}
```
public AsposeInvalidDataException(String message, Throwable innerException)
```


Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a specified error message and a reference to the inner exception that is the cause of this exception.

**Parameters:**
| Parameter | Type | Description |
| --- | --- | --- |
| message | java.lang.String | The error message that explains the reason for the exception. |
| innerException | java.lang.Throwable | The exception that is the cause of the current exception, or a null reference (Nothing in Visual Basic) if no inner exception is specified. |

### AsposeInvalidDataException(Throwable innerException) {#AsposeInvalidDataException-java.lang.Throwable-}
```
public AsposeInvalidDataException(Throwable innerException)
```


Initializes a new instance of the [AsposeInvalidDataException](../../com.aspose.email/asposeinvaliddataexception) class with a reference to the inner exception that is the cause of this exception.

**Parameters:**
| Parameter | Type | Description |
| --- | --- | --- |
| innerException | java.lang.Throwable | The exception that is the cause of the current exception, or a null reference (Nothing in Visual Basic) if no inner exception is specified. |

